lib/gc/Transforms/Pipeline.cpp

```diff
@@ -149,10 +151,45 @@ void populateCPUPipeline(mlir::OpPassManager &pm) {
   populateLLVMPasses(pm);
 }
 
+void populateGPUPipeline(mlir::OpPassManager &pm) {
+  pm.addNestedPass<func::FuncOp>(createLinalgGeneralizeNamedOpsPass());
+  bufferization::OneShotBufferizationOptions options;
+  options.bufferizeFunctionBoundaries = true;
+  options.setFunctionBoundaryTypeConversion(
+      bufferization::LayoutMapOption::IdentityLayoutMap);
+  pm.addPass(bufferization::createOneShotBufferizePass(options));
+  pm.addPass(createCSEPass());
+  pm.addNestedPass<func::FuncOp>(createConvertLinalgToParallelLoopsPass());
+  pm.addNestedPass<func::FuncOp>(createGpuMapParallelLoopsPass());
+  pm.addNestedPass<func::FuncOp>(createParallelLoopToGpuPass());
+  pm.addNestedPass<func::FuncOp>(createLowerAffinePass());
+  pm.addPass(memref::createNormalizeMemRefsPass());
+  pm.addPass(createGpuKernelOutliningPass());
+  pm.addPass(createGpuLegalizeModule());
+  pm.addPass(memref::createFoldMemRefAliasOpsPass());
+  ConvertIndexToLLVMPassOptions idxOptions;
+  idxOptions.indexBitwidth = 32;
+  pm.addNestedPass<gpu::GPUModuleOp>(createConvertIndexToLLVMPass(idxOptions));
+  pm.addNestedPass<gpu::GPUModuleOp>(createConvertGpuOpsToLLVMSPVOps());
+  pm.addPass(createCanonicalizerPass());
+  pm.addNestedPass<gpu::GPUModuleOp>(createConvertGpuSignaturesToLLVM());
+  pm.addPass(createGpuToLLVMConversionPass());
+  pm.addPass(createCanonicalizerPass());
+  pm.addPass(createGpuGenAttachTarget());
+  GpuModuleToBinaryPassOptions gpuModuleToBinaryPassOptions;
+  pm.addPass(createGpuModuleToBinaryPass(gpuModuleToBinaryPassOptions));
```

Comment:
Which tests did you use to verify the pipeline?

If I try to run the GPU pipeline on [this simple matmul test](https://github.com/dchigarev/graph-compiler/blob/fix-linalg-to-xe/test/mlir/test/gc/gpu-runner/XeGPU/f16_matmul_32x32.mlir) the `gpu-module-to-binary` pass fails with the following error:
```
error: LLVM Translation failed for operation: builtin.unrealized_conversion_cast
```

don't we need to add `reconcile-unrealized-casts` somewhere?

Reply:
Testing it on a simple vector add for now. I'll add it once we can execute it. Yes, we will need the reconcile pass for the pipeline to be complete. I'd like to get to an end-to-end working scenario first though.